### PR TITLE
Type cleanup

### PR DESCRIPTION
**Pull request by fredj**
All these error have been found while I was working on trying to simplify `externs/olx.js` by using `@record` instead of `@typedef` (see https://github.com/google/closure-compiler/wiki/Structural-Interfaces-in-Closure-Compiler)

### CODE REVIEW

**Comment by fredj** [comment]
Thanks for any review.

File: src/ol/tilegrid/tilegrid.js [hunk 1]
```diff
@@ -541,9 +541,7 @@ ol.tilegrid.createForExtent = function(extent, opt_maxZoom, opt_tileSize, opt_co
  * @api
  */
 ol.tilegrid.createXYZ = function(opt_options) {
-  var options = /** @type {olx.tilegrid.TileGridOptions} */ ({});
-  ol.object.assign(options, opt_options !== undefined ?
-      opt_options : /** @type {olx.tilegrid.XYZOptions} */ ({}));
+  var options = ol.object.assign({}, opt_options ? opt_options : {});
```
Comment (gberaudo) [inline src/ol/tilegrid/tilegrid.js]: Shouldn't `options` be casted to `olx.tilegrid.TileGridOptions` at this point?

Comment (fredj) [inline src/ol/tilegrid/tilegrid.js]: `options` can't be a `olx.tilegrid.TileGridOptions` because it has the `maxZoom` property

**Comment by fredj** [comment]
Any other review?

**Comment by petrsloup** [comment]
I think all these changes make sense. LGTM

**Comment by fredj** [comment]
Thanks for the review

**Pull request merged**